src/Servers/HttpSys/test/FunctionalTests/DelegateTests.cs

```diff
@@ -3,16 +3,35 @@
 
 using System.Net.Http;
 using System.Runtime.InteropServices;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Hosting.Server;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.HttpSys.Internal;
 using Microsoft.AspNetCore.Testing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
 
 namespace Microsoft.AspNetCore.Server.HttpSys.FunctionalTests;
 
 public class DelegateTests
 {
     private static readonly string _expectedResponseString = "Hello from delegatee";
 
+    [ConditionalFact]
+    [DelegateSupportedCondition(true)]
+    public void IServerDelegationFeature_IsAvailableFromServices()
+    {
+        var builder = new HostBuilder();
+        builder.ConfigureWebHost(webHost =>
+        {
+            webHost.UseHttpSys();
+        });
+        using var host = builder.Build();
+        var server = host.Services.GetRequiredService<IServer>();
+        var delegationFeature = host.Services.GetRequiredService<IServerDelegationFeature>();
+        Assert.Same(server, delegationFeature);
```

Comment:
Isn't this more an implantation detail that a strict requirement?

Reply:
Yes, I just wanted to make sure they didn't create separate server instances.

Reply:
Makes sense. I generally error on the side of being too prescriptive in unit tests anyway. There's a limit to this, but if the implementation details change and we can convince ourselves it's non-breaking, we can always just update the tests.